File: Java/Problem_025.java
package ProjectEuler;

import java.math.BigInteger;

public class Problem_025 {

    public static void main(String[] args) {
        double start = System.nanoTime();

        BigInteger a = BigInteger.ONE;
        BigInteger b = BigInteger.ONE;
        BigInteger temp;

        for (int i = 3; ; i++) {
            temp = a.add(b);
            a = b;
            b = temp;
            if (temp.toString().length() == 1000) {
                System.out.println(i);
                break;
            }
        }

        System.out.println((System.nanoTime() - start) / 1_000_000 + " ms");
    }

}

File: Python/Problem 025.py
from time import clock

start = clock()

digits = 1000

a = 1
b = 1
temp = 0
i = 2

while True:
    temp = a + b
    a = b
    b = temp
    i += 1
    if len(str(b)) == digits:
        print(i)
        break

print(clock() - start)